apollo-coroutines-support/src/main/kotlin/com/apollographql/apollo/coroutines/CoroutinesExtensions.kt

```diff
@@ -73,35 +73,27 @@ fun <T> ApolloQueryWatcher<T>.toFlow(): Flow<Response<T>> = callbackFlow {
 
 
 /**
- * Converts an [ApolloCall] to an [Deferred]. This is a convenience method that will only return the first value emitted.
- * If the more than one response is required, for an example to retrieve cached and network response, use [toFlow] instead.
+ * Suspends the [ApolloCall] until it completes and returns the value on success or throws an exception on failure.
+ * The [ApolloCall] is cancelled when the coroutine running the operation is cancelled as well.
  *
  * @param <T>  the value type.
- * @return the deferred
+ * @return the response on success.
+ * @throws ApolloException on failure.
  */
-fun <T> ApolloCall<T>.toDeferred(): Deferred<Response<T>> {
-  val deferred = CompletableDeferred<Response<T>>()
-
-  deferred.invokeOnCompletion {
-    if (deferred.isCancelled) {
-      cancel()
-    }
-  }
+suspend fun <T> ApolloCall<T>.await(): Response<T> = suspendCancellableCoroutine { cont ->
   enqueue(object : ApolloCall.Callback<T>() {
     override fun onResponse(response: Response<T>) {
-      if (deferred.isActive) {
-        deferred.complete(response)
-      }
+      cont.resume(response)
```

Comment:
You should have some kind of boolean to check if you already resumed once, otherwise an exception will be thrown second time the callback is called (I assume it's possible)

Reply:
Thanks for your notice, I've adapted the code a bit, hope it's now